src/components/Profile/components/SchedulePanel/index.jsx

```diff
@@ -42,408 +40,212 @@ const styles2 = {
-    this.reloadHandler = this.reloadHandler.bind(this);
-  }
-
-  componentDidMount() {
-    this.loadData(this.props.profile);
-  }
-
-  loadData = async (searchedUser) => {
+const GordonSchedulePanel = (props) => {
+  const [myProf, setMyProf] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(props, myProf ? false : true);
+  const [disabled, setDisabled] = useState(true);
+  const [selectedEvent, setSelectedEvent] = useState(null);
+  const [isDoubleClick, setIsDoubleClick] = useState(false);
+  const [description, setDescription] = useState('');
+  const [modifiedTimeStamp, setModifiedTimeStamp] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [resourceId, setResourceId] = useState(0);
+  const [reloadCall, setReloadCall] = useState(false);
+  const [editDescriptionOpen, setEditDescriptionOpen] = useState(false);
+  const [scheduleControlInfo, setScheduleControlInfo] = useState(null);
+  // Remove the line: let scheduleControlInfo = null;
+
+  useEffect(() => {
+    loadData(props.profile);
+  }, [props.profile]);
+
+  const loadData = async (searchedUser) => {
     try {
       const scheduleControlInfo = await schedulecontrol.getScheduleControl(
         searchedUser.AD_Username,
       );
-      this.scheduleControlInfo = scheduleControlInfo;
-    } catch (e) {
-      this.setState({ loading: false });
-    }
-    if (this.scheduleControlInfo) {
-      this.setState({
-        isSchedulePrivate: this.scheduleControlInfo.IsSchedulePrivate,
-        description: this.scheduleControlInfo.Description
-          ? this.scheduleControlInfo.Description.replace(new RegExp('SlSh', 'g'), '/')
-              .replace(new RegExp('CoLn', 'g'), ':')
-              .replace(new RegExp('dOT', 'g'), '.')
-          : '',
-        modifiedTimeStamp: this.scheduleControlInfo.ModifiedTimeStamp,
-      });
-    }
-    this.setState({ loading: false });
-  };
-
-  handleMyScheduleOpen = (slotInfo) => {
-    if (this.props.myProf) {
-      this.setState({ myScheduleOpen: true });
-      if (slotInfo) {
-        let startTime = slotInfo.start.toTimeString().split(':');
-        let endTime = slotInfo.end.toTimeString().split(':');
-        this.setState({
-          start: startTime[0] + ':' + startTime[1],
-          end: endTime[0] + ':' + endTime[1],
-          resourceId: slotInfo.resourceId,
+      if (scheduleControlInfo) {
+        setScheduleControlInfo({
+          isSchedulePrivate: scheduleControlInfo.IsSchedulePrivate,
+          description: scheduleControlInfo.Description
+            ? scheduleControlInfo.Description.replace(new RegExp('SlSh', 'g'), '/')
+                .replace(new RegExp('CoLn', 'g'), ':')
+                .replace(new RegExp('dOT', 'g'), '.')
+            : '',
+          modifiedTimeStamp: scheduleControlInfo.ModifiedTimeStamp,
         });
       }
-      this.setState({ isDoubleClick: false });
+    } catch (e) {
+      setLoading(false);
     }
+    setLoading(false);
```

Comment:
```suggestion
    } catch (e) {

    }
    setLoading(false);
```

The first set loading is redundant